sphinx_immaterial/apidoc/cpp/api_parser.py

```diff
@@ -429,18 +441,57 @@ def _get_all_decls(config: Config, cursor: Cursor, allow_file):
             yield from _get_all_decls(config, child, None)
 
 
+def split_doc_comment_into_lines(cmt: str) -> List[str]:
+    """Strip the raw string of an object's comment into lines.
+    :param cmt: the comment to parse.
+    :returns: A list of the lines without the surrounding C++ comment syntax.
+    """
+    # the first line is never indented (in `raw_comment` form)
+    # so dedent all subsequent lines only
+    first_line_end = cmt.find("\n")
+    cmt = cmt[:first_line_end] + dedent(cmt[first_line_end:])
+    # split into a list of lines & account for CRLF and LF line endings
+    body = [line.rstrip("\r") for line in cmt.splitlines()]
+
+    # strip all the comment syntax out
+    if body[0].startswith("//"):
+        body = [line.lstrip("/").lstrip("!").lstrip("<") for line in body]
+    elif body[0].startswith("/*"):
+        body[0] = body[0].lstrip("/").lstrip("*").lstrip("!")
+        multi_lined_asterisk = True  # works also for single-line comments blocks
+        if len(body) > 1:
+            line_has_asterisk = [line.startswith("*") for line in body[1:]]
+            multi_lined_asterisk = line_has_asterisk.count(True) == len(body) - 1
+        body = [
+            (line.lstrip("*").lstrip("<") if multi_lined_asterisk else line)
+            for line in body
+        ]
+        body[-1] = body[-1].rstrip("*/").rstrip()
+    if len(body) > 1:
+        body = dedent("\n".join(body)).splitlines()
+    return [""] if not body else body
+
+
+DOC_COMMENT_PREFIX = re.compile(r"/(//|/\!|\*\*|\*\!)\<?")
+
+
 def get_doc_comment(config: Config, cursor: Cursor):
     translation_unit = cursor.translation_unit
     for token in cursor.get_tokens():
         location = token.location
         break
     else:
         location = cursor.location
-    initial_location = location
     f = location.file
     line = location.line
-    presumed_file, presumed_line, presumed_column = get_presumed_location(location)
     end_location = SourceLocation.from_position(translation_unit, f, line, 1)
+    if cursor.raw_comment and DOC_COMMENT_PREFIX.match(cursor.raw_comment):
```

Comment:
Is there any case where we need the old logic?  Otherwise we may as well remove it.

Reply:
Not that I know of right now. I can remove it, and if we need it in the future, then we can fetch it from the git history.

Reply:
removed old logic.

I also removed the regex `DOC_COMMENT_PREFIX` because the new `NON_DOC_COMMENT` regex is used to remove any non-docstring comments from within the given `raw_comment`.